mintpy/smallbaselineApp.py

```diff
@@ -1063,8 +1065,72 @@ def plot_result(self, print_aux=True, plot=True):
+            tropo_model = self.template['mintpy.troposphericDelay.weatherModel']
+            geo_dir = os.path.join(self.workDir, 'geo')
+
+            # view options
+            opt1 = ['--nodisplay', '--dpi', '150', '--update']
+            opt2 = ['--nodisplay', '--dpi', '150', '--update', '--mask', mask_file, '--noaxis', '-u', 'cm', '--wrap-range', '-10', '10']
+
+            iargs_list = [
+                ['velocity.h5'] + opt1,
+                ['temporalCoherence.h5'] + opt1 + ['-c', 'gray', '--vlim', '0', '1'],
+                ['maskTempCoh.h5'] + opt1 + ['-c', 'gray', '--vlim', '0', '1'],
+                [dem_file] + opt1,
+                [ifgram_stack] + opt1 + ['-unwrapPhase', '--zero-mask', '--wrap'],
+                [ifgram_stack] + opt1 + ['-unwrapPhase', '--zero-mask'],
+                # [ifgram_stack] + opt1 + ['-coherence', '--mask', 'no'],
+                # [ifgram_stack] + opt1 + ['-connectComponent', '--mask', 'no'],
+                [ifgram_stack] + opt1 + ['-unwrapPhase_bridging', '--zero-mask'],
+                [ifgram_stack] + opt1 + ['-unwrapPhase_closure', '--zero-mask'],
+                [ifgram_stack] + opt1 + ['-unwrapPhase_bridging_phaseClosure', '--zero-mask'],
+                ['avgPhaseVelocity.h5'] + opt1,
+                ['avgSpatialCoh.h5'] + opt1 + ['-c', 'gray', '--vlim', '0', '1'],
+                ['maskConnComp.h5'] + opt1 + ['-c', 'gray', '--vlim', '0', '1'],
+                ['timeseries.h5'] + opt2,
+                ['timeseries_LODcor.h5'] + opt2,
+                ['timeseries_LODcor_ECMWF.h5'] + opt2,
+                ['timeseries_LODcor_ECMWF_demErr.h5'] + opt2,
+                ['timeseries_LODcor_ECMWF_ramp.h5'] + opt2,
+                ['timeseries_LODcor_ECMWF_ramp_demErr.h5'] + opt2,
+                [f'timeseries_{tropo_model}.h5'] + opt2,
+                [f'timeseries_{tropo_model}_demErr.h5'] + opt2,
+                [f'timeseries_{tropo_model}_ramp.h5'] + opt2,
+                [f'timeseries_{tropo_model}_ramp_demErr.h5'] + opt2,
+                [f'timeseries_ramp.h5'] + opt2,
+                [f'timeseries_demErr_ramp.h5'] + opt2,
+                [f'timeseries_demErr.h5'] + opt2,
+                [os.path.join(geo_dir, 'geo_maskTempCoh.h5')] + opt1 + ['-c', 'gray'],
+                [os.path.join(geo_dir, 'geo_temporalCoherence.h5')] + opt1 + ['-c', 'gray'],
+                [os.path.join(geo_dir, 'geo_velocity.h5')] + opt1 + ['velocity'],
+                [os.path.join(geo_dir, 'geo_timeseries_ECMWF_demErr_ramp.h5')] + opt1 + ['--noaxis'],
+                [os.path.join(geo_dir, 'geo_timeseries_ECMWF_demErr.h5')] + opt1 + ['--noaxis'],
+                [os.path.join(geo_dir, 'geo_timeseries_demErr_ramp.h5')] + opt1 + ['--noaxis'],
+                [os.path.join(geo_dir, 'geo_timeseries_demErr.h5')] + opt1 + ['--noaxis'],
+                [f'velocity{tropo_model}.h5'] + opt1 + ['--mask', 'no'],
+                [f'numInvIfgram.h5'] + opt1 + ['--mask', 'no']
+            ]
+
```

Comment:
And at here, loop over the iarg_list, within each loop, if * is in the file path, run glob.glob() to search all existing files, and replace the original element list with multiple element lists: one for each existing files.

Reply:
> `unwrapPhase`, `coherence`, `connectComponent` and `velcoity` are all datasets and should be immediately after the input file such as `view.py velocity.h5 velocity` without - or --.

Thanks. This has fixed the errors except for `connectComponent`. It shows this error. I'm not sure if it's just related to my sample dataset missing this dataset.

```python
Exception: No input dataset found!
available datasets:
['unwrapPhase-20200109_20200121', 'unwrapPhase-20200109_20200226', 'unwrapPhase-20200109_20200309', 'unwrapPhase-20200121_20200226', 'unwrapPhase-20200121_20200309', 'unwrapPhase-20200121_20200321',
'unwrapPhase-20200226_20200309', 'unwrapPhase-20200226_20200321', 'unwrapPhase-20200226_20200402', 'unwrapPhase-20200309_20200321', 'unwrapPhase-20200309_20200402', 'unwrapPhase-20200309_20200414',
'coherence-20200109_20200121', 'coherence-20200109_20200226', 'coherence-20200109_20200309',
'coherence-20200121_20200226', 'coherence-20200121_20200309', 'coherence-20200121_20200321', 'coherence-20200226_20200309', 'coherence-20200226_20200321', 'coherence-20200226_20200402', 'coherence-20200309_20200321', 'coherence-20200309_20200402', 'coherence-20200309_20200414',
'wrapPhase-20200109_20200121', 'wrapPhase-20200109_20200226', 'wrapPhase-20200109_20200309', 'wrapPhase-20200121_20200226', 'wrapPhase-20200121_20200309', 'wrapPhase-20200121_20200321', 'wrapPhase-20200226_20200309', 'wrapPhase-20200226_20200321', 'wrapPhase-20200226_20200402', 'wrapPhase-20200309_20200321', 'wrapPhase-20200309_20200402', 'wrapPhase-20200309_20200414']
```

Reply:
Currently not all supported InSAR processors provide the connectComponent data, so it's missing in your test. This only happens to the `ifgramStack.h5` file. In the .sh file, I used `h5ls` to check if the dataset exists; in python, you could check as below:

```python
if 'connectComponent' in readfile.get_dataset_list(fname):
```